Adding search bar component to Hero

ajolipa opened the pull request:
### In this PR
Adds a `SearchBar` MDX component for use inside rich text components, e.g. in page heroes.
<img width="2301" height="369" alt="image" src="https://github.com/user-attachments/assets/c7fc5481-2367-45d8-bb95-45d4c867f8d6" />

### Questions and Notes
- How configurable will we need things like the button color to be?
- I think ideally the `SearchBar` component would receive an `onSubmit` prop rather than forcing the submit action to be redirecting to a URL with the query string appended; however, with the way the data comes in from Tina, and with the way that Astro handles MDX components inside an Astro component it was much easier to just have a script add the `submit` method rather than trying to use React hooks. Open to suggestions on configuring that differently though.

dleadbetter commented on src/apps/pages/sections/mdx/SearchBar.tsx:
```diff
@@ -0,0 +1,37 @@
+}
+
+const SearchBar = (props: SearchBarProps) => {
+  const { button_text, placeholder, url } = props;
```
Could we rename this to camel case?

`const { button_text: buttonText } ....`

dleadbetter commented on src/apps/pages/sections/FreeText.astro:
```diff
@@ -9,5 +10,21 @@ const { content } = Astro.props;
\ No newline at end of file
+</article>
+
+<script>
```
This seems like an awkward place for this logic. The FreeText component isn't guaranteed to have any search bars and really shouldn't have any knowledge of the MDX child components. Is there any reason we wouldn't want this in the `SearchBar` component?

dleadbetter replied:
I think if we're having the component be interactive on the client, we're going to have to allow for some client rendering. Otherwise we'll end up with Javascript business logic disconnected from the presentation, which is going to make the code hard to follow. It would be much cleaner to use React's state management to handle the input value and define a function for `onSubmit`. I think you should be able to use the `client:load` directive to still render the component on the server and have it hydrate on the client.

ajolipa replied:
So you mean option 3 from the list above? Make the whole `free_text` section (and anywhere else we might use the search bar) `client:load`?

dleadbetter commented on src/apps/pages/sections/FreeText.astro:
```diff
@@ -9,5 +10,21 @@ const { content } = Astro.props;
 >
   <TinaMarkdown
     content={content}
+    components={richTextComponents}
```
Interesting pull this out into a utility class. Could you explain a little about why we didn't just inline the component?

ajolipa commented on src/apps/pages/sections/mdx/SearchBar.tsx:
```diff
@@ -0,0 +1,37 @@
+import { Icon } from '@performant-software/core-data/ssr';
```
I guess I just envisioned keeping the MDX components used on the custom pages all together with the section components, but I'm not attached to that.

ajolipa commented on tina/content/pages.ts:
```diff
@@ -29,6 +29,25 @@ const SpacerSizes = [{
+    name: 'url',
+    label: 'Search URL',
+    type: 'string',
+    description: 'Should be the full URL excluding only the query string itself, e.g. `https:://mysite.com/search?index[query]=`.'
```
If the format will be consistent for all pages; for the map searches it looks like the format is actually `?q=`, but I noticed on NBU that for the list searches (people, archives) that the format is `?people[query]=` etc. And actually it looks like currently the list search implementation on CDP isn't using URL parameters? So I guess that would need to be changed for this approach to work at all. 🤔

ajolipa commented:
UPDATE: I've reworked this a bit; I don't think it's necessary or quite makes sense to have it as an MDX component, given how the markdown renderer works and where it shows up in the designs, so I've changed it to a regular React component and added it to the `Hero` section for now. As other content sections are implemented, now that the component exists, it can be added there as well (e.g. I intend for it to be a possible template for the centered text with background component).
